CMakeLists.txt

```diff
@@ -139,7 +139,7 @@ if(CMAKE_COMPILER_IS_GNU)
     # note: starting with CMake 2.8 we could use CMAKE_C_COMPILER_VERSION
     execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion
                     OUTPUT_VARIABLE GCC_VERSION)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wwrite-strings")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wno-type-limits -Wwrite-strings")
```

Comment:
As a general rule, please do not disable compiler warnings. When newer compiler versions introduce new warnings that trigger on our code, we generally prefer to fix the code to avoid those warnings.

Looking at the offending code that you mention in the commit message for 69252ab39a39dfb6f1ba686ab92fa56e58b21e94, this is a genuine bug. Thanks for noticing it, but it needs to be fixed, not ignored.